syntax = "proto3";

package user.grpc;

import "google/protobuf/empty.proto";

service User {
    rpc getUsers(google.protobuf.Empty) returns (UsersResponse) {}
    rpc createUser(createUserRequest) returns (createUserResponse) {}
}

message UserResponse {
    int64  id = 1;
    string first_name = 2;
    string last_name = 3;
    string email = 4;
    string gender = 5;
}

message UsersResponse {
    repeated UserResponse users = 1;
}

message createUserRequest {
    string first_name = 1;
    string last_name = 2;
    string email = 3;
    string gender = 4;
}

message createUserResponse {
    string result = 1;
}
